app.py

```diff
@@ -28,18 +24,72 @@ def inject_func():
                 len=len)
 
 
-## initiate ###################
+###### before and after each request ########
+@app.before_request
+def open_conn():
+    global connections
+    conn = connections.getconn()
+    cursor = conn.cursor(cursor_factory = DictCursor)
+    g.cur = cursor
+    g.conn = conn
+    g.error = False
+
+
+@app.after_request
+def close_conn(response):
+    
+    if g.conn is not None:    
+        conn = g.conn
+        cur = g.cur
+        conn.commit()
+        cur.close()
+        global connections
+        connections.putconn(conn)
+        return response
+    else:
+        return response
+
+
+########### Error handler ###########
+@app.errorhandler(DataError)
```

Comment:
What is `DataError`?

Reply:
https://www.psycopg.org/docs/errors.html

but my real intention was checking if the stacking works

Reply:
Ah ok so we need it. thanks.